### PR TITLE
Enable overflow by default

### PR DESCRIPTION
**Pull request by bcardiff**
This PR makes the overflow behavior active by default.

A temporal flag `disable_overflow` is added in case this change cause stopper issues. Expect to be removed in the near future.

Instead of linking to compiler-rt, the existing minimal port will be used. The specs of `mulodi4` were ported also.

The `mulodi4` implementation was moved to a `compiler_rt` directory. More compiler-rt functions might be ported if we aim for Int128 bit support across all targets. But that is a different story.

The ported `mulodi4` is not always used (it depends on the architecture) but is always emitted. If the user wants to link an official compiler-rt, the `skip_crystal_compiler_rt` flag can be used. But I expect the port will be easier to use.

cc: @wontruefree
Ref: #7514, #8153, #7798

### CODE REVIEW

File: spec/compiler/codegen/and_spec.cr [hunk 1]
```diff
@@ -18,11 +18,11 @@ describe "Code gen: and" do
   end
 
   it "codegens and with bool and int 1" do
-    run("struct Bool; def to_i; 0; end; end; (false && 2).to_i").to_i.should eq(0)
+    run("struct Bool; def to_i!; 0; end; end; (false && 2).to_i!").to_i.should eq(0)
```
Comment (asterite) [inline spec/compiler/codegen/and_spec.cr]: Is this rename really needed? It's just a spec, I don't think it breaks anything if we use `to_i`, right?

Comment (asterite) [reply spec/compiler/codegen/and_spec.cr]: Oh, nevermind, it does because we later invoke it on `2`.

Comment (bcardiff) [reply spec/compiler/codegen/and_spec.cr]: Yeah, ideally we should be able to use exceptions with the JIT runner that does not include prelude. But that is a pending thing to be done. But the specs test more the dispatch and typing than the actual behavior.

**Comment by wontruefree** [comment]
Thanks for the CC.  This broke my build and it was nice to find the reason why.
